### ISSUE TITLE
eCommerce Plan: Upgrading From Business Switches the Theme to Storefront

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/wc-calypso-bridge#572]
#### Steps to reproduce

1. Starting at URL: https://wordpress.com/plans/{SITEURL}?customerType=business
2. Upgrade from the Business plan to eCommerce
3. The active theme is changed to Storefront without any notification.

#### What I expected

The current theme to remain active

#### What happened instead

Storefront was activated without any user-facing notification.

#### Browser / OS version

Chrome / Mac OS 

#### Context / Source

We were contacted in #12438970-hc. User had updated from Business to eCommerce, and the site very visibly changed. It had been using a third-party theme that made use of a page builder and switching away from that theme changed not only the site's layout, but also caused the shortcodes from the page builder to display.

The user was concerned, and initially asked to return to the Business plan, because it wasn't clear to them that it was a theme change causing the issue. Switching back to their previous theme corrected the issue. 

As a side note, there was also no logging of the theme change. The Activity Log didn't show it at all. 

I can see why we may activate Storefront for a new purchase, but for a site that is already setup using another theme, having it changed unannounced can create confusion.

#user-report

**Comment by retnonindya** [comment]
User report: 22184233-hc

The user is understandably in distress when their theme changed drastically without them knowing as soon as they upgrade to eCommerce plan. We rewound the site to the point before eCommerce plan purchased and switched back to Business plan.

**Comment by eurello** [comment]
@timmyc do you have any ETA of when this bug might be looked into?

**Comment by timmyc** [comment]
@eurello this honestly is a bit out of our area of expertise with the ecommerce plan - as this bug is part of the provisioning process that happens on WordPress.com - where this repo is more about the WooCommerce UI side of things of the ecommerce plan. I did open up a phabricator diff last week that I think will fix it: https://code.a8c.com/D46075 but will definitely need someone more familiar with that code to review. 

I'll ping in the atomic channel to see if someone could take a look at the proposed fix.

**Comment by timmyc** [comment]
Reached out for a review/advice from the Atomic team here: p9o2xV-Wz-p2

**Comment by eurello** [comment]
Thanks so much!

**Comment by timmyc** [comment]
Fix deployed in r210691-wpcom. I just tested it out the fix in production with a new business plan site. Upgraded to the ecommerce plan and verified my prior custom theme choice on the biz plan site was still present and active. Here is what the theme step in the Woo onboarding wizard looked like with the fix in place:

![Home ‹ WooCommerce ‹ 💯 — WooCommerce 2020-07-20 08-18-18](https://user-images.githubusercontent.com/22080/87954892-d1e05580-ca61-11ea-97ed-7e8074402816.png)

Closing this out.
